File: Python Codes/SwapFirstLastDig.py
import  math
num = int(input("Enter a number: "))
ldig = num%10 # Finding last dig
n = int(math.log10(num)) + 1 # Number of digits
fdig = int(num//(10**(n-1))) # finding first digit by dividing the number by 10^(n-1)
swappedNum = ldig*(10**(n-1)) # Placing ldig in place of fdig
swappedNum += num%(10**(n-1)) # adding the remaining number
swappedNum -= ldig # removing last dig
swappedNum += fdig # adding fdig as last digit
print(swappedNum)

File: C program/New Batch/SwapFirstLastDig.cpp
#include<stdio.h>
int pow(int b,int e){
    int ans=1;
    for(int i=0;i<e;i++){
        ans *= b;
    }
    return ans;
}
int main(){
    int n;
    printf("Enter a number: ");
    scanf("%d",&n);
    int num = n;
    int fdig;
    int ldig = num%10;
    while(num>0){
        fdig = num%10;
        num /= 10;
    }
    int ans = 0;
    ans = fdig;
    n /= 10;
    int dig;//2
    int i=1;
    while(n>10){
        dig = n%10;
        ans = dig*pow(10,i++) + ans;
        n /= 10;
    }
    ans = ldig*pow(10,i)+ans;
    printf("Answer: %d",ans);
    return 0;
}